osu.Game/Graphics/UserInterfaceV2/IFormControl.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Game.Graphics.UserInterfaceV2
+{
+    /// <summary>
+    /// Represents an interface for all form controls.
+    /// </summary>
+    public interface IFormControl : IDrawable, IHasFilterTerms
```

Comment:
Why does this have to implement `IHasFilterTerms`? I have vague unsubstantiated reservations about doing this, I'd like that to be handled by `SettingsItemV2` if possible.

Does this mean that if a settings item wants to specify filter terms that are not specifically tied to the caption or hint text then it will be required to subclass the form control? I would not want that. I would want this to be declarative.

Reply:
> Why does this have to implement IHasFilterTerms? I have vague unsubstantiated reservations about doing this, I'd like that to be handled by SettingsItemV2 if possible.

Filter terms vary based on the control. For a dropdown particularly, it includes the name of each dropdown menu item. If this looks too much out of place, it can be exposed via a different property like `Terms` or `Keywords`, anything that removes the idea of "filtering" from form controls.

> Does this mean that if a settings item wants to specify filter terms that are not specifically tied to the caption or hint text then it will be required to subclass the form control? I would not want that. I would want this to be declarative.

There is a `Keywords` property exposed for that purpose.

Reply:
I forgot that `IHasFilterTerms` is not the same thing as `IFilterable`, and the second one is the one that is magicked upon by `SearchContainer` descending the tree to do its filtering magic.

This is probably fine to stay.